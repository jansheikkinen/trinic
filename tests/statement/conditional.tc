// Conditional

function main(void) -> void do
  if 1 do print(1); end
  if 1 do print(1); else print(2); end
  if 1 do print(1); else if 2 do print(2); end
  if 1 do print(1); else if 2 do print(2); else print(3); end

  if 1 do
    if 2 do
      print(1 + 2);
    else
      print(3 + 4);
    end
  end
end
