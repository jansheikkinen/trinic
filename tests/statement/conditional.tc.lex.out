
### LEXER ###
000:     FUNCTION,     (null), (003, 008)
001:   IDENTIFIER,       main, (003, 013)
002:            (,     (null), (003, 014)
003:         VOID,     (null), (003, 018)
004:            ),     (null), (003, 019)
005:           ->,     (null), (003, 021)
006:         VOID,     (null), (003, 026)
007:           DO,     (null), (003, 029)
008:           IF,     (null), (004, 004)
009:      INTEGER,          1, (004, 007)
010:           DO,     (null), (004, 009)
011:        PRINT,     (null), (004, 015)
012:            (,     (null), (004, 016)
013:      INTEGER,          1, (004, 018)
014:            ),     (null), (004, 018)
015:            ;,     (null), (004, 019)
016:          END,     (null), (004, 023)
017:           IF,     (null), (005, 004)
018:      INTEGER,          1, (005, 007)
019:           DO,     (null), (005, 009)
020:        PRINT,     (null), (005, 015)
021:            (,     (null), (005, 016)
022:      INTEGER,          1, (005, 018)
023:            ),     (null), (005, 018)
024:            ;,     (null), (005, 019)
025:         ELSE,     (null), (005, 024)
026:        PRINT,     (null), (005, 030)
027:            (,     (null), (005, 031)
028:      INTEGER,          2, (005, 033)
029:            ),     (null), (005, 033)
030:            ;,     (null), (005, 034)
031:          END,     (null), (005, 038)
032:           IF,     (null), (006, 004)
033:      INTEGER,          1, (006, 007)
034:           DO,     (null), (006, 009)
035:        PRINT,     (null), (006, 015)
036:            (,     (null), (006, 016)
037:      INTEGER,          1, (006, 018)
038:            ),     (null), (006, 018)
039:            ;,     (null), (006, 019)
040:         ELSE,     (null), (006, 024)
041:           IF,     (null), (006, 027)
042:      INTEGER,          2, (006, 030)
043:           DO,     (null), (006, 032)
044:        PRINT,     (null), (006, 038)
045:            (,     (null), (006, 039)
046:      INTEGER,          2, (006, 041)
047:            ),     (null), (006, 041)
048:            ;,     (null), (006, 042)
049:          END,     (null), (006, 046)
050:           IF,     (null), (007, 004)
051:      INTEGER,          1, (007, 007)
052:           DO,     (null), (007, 009)
053:        PRINT,     (null), (007, 015)
054:            (,     (null), (007, 016)
055:      INTEGER,          1, (007, 018)
056:            ),     (null), (007, 018)
057:            ;,     (null), (007, 019)
058:         ELSE,     (null), (007, 024)
059:           IF,     (null), (007, 027)
060:      INTEGER,          2, (007, 030)
061:           DO,     (null), (007, 032)
062:        PRINT,     (null), (007, 038)
063:            (,     (null), (007, 039)
064:      INTEGER,          2, (007, 041)
065:            ),     (null), (007, 041)
066:            ;,     (null), (007, 042)
067:         ELSE,     (null), (007, 047)
068:        PRINT,     (null), (007, 053)
069:            (,     (null), (007, 054)
070:      INTEGER,          3, (007, 056)
071:            ),     (null), (007, 056)
072:            ;,     (null), (007, 057)
073:          END,     (null), (007, 061)
074:           IF,     (null), (009, 004)
075:      INTEGER,          1, (009, 007)
076:           DO,     (null), (009, 009)
077:           IF,     (null), (010, 006)
078:      INTEGER,          2, (010, 009)
079:           DO,     (null), (010, 011)
080:        PRINT,     (null), (011, 011)
081:            (,     (null), (011, 012)
082:      INTEGER,          1, (011, 014)
083:            +,     (null), (011, 015)
084:      INTEGER,          2, (011, 018)
085:            ),     (null), (011, 018)
086:            ;,     (null), (011, 019)
087:         ELSE,     (null), (012, 008)
088:        PRINT,     (null), (013, 011)
089:            (,     (null), (013, 012)
090:      INTEGER,          3, (013, 014)
091:            +,     (null), (013, 015)
092:      INTEGER,          4, (013, 018)
093:            ),     (null), (013, 018)
094:            ;,     (null), (013, 019)
095:          END,     (null), (014, 007)
096:          END,     (null), (015, 005)
097:          END,     (null), (016, 003)
